.github/workflows/task-test.yml

```diff
@@ -80,26 +75,61 @@ jobs:
             tests/pytests/requirements.*
       - name: Deps checkout (node20 not supported)
         if: steps.node20.outputs.supported == 'false'
-        uses: actions/checkout@v3
-        with:
-          path: setup
-          sparse-checkout-cone-mode: false
-          sparse-checkout: |
-            .install
-            tests/pytests/requirements.*
-      - name: Setup specific
+        run: |
+          # Execute the logic based on the detected platform
```

Comment:
If this snippet is repeated, consider putting it in a bash script and reuse it?

Reply:
You could use a separate repo to house your custom actions and expose them as a github action.